src/Driver/Mysqli/HostnameRequired.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Driver\Mysqli;
+
+final class HostnameRequired extends MysqliException
+{
+    public static function fromPersistentConnection() : self
+    {
+        return new self('Hostname is required for persistent connection');
```

Comment:
I'd go with either
```suggestion
        return new self('The host parameter is required for persistent connections');
```

or

```suggestion
        return new self('host parameter required for persistent connection');
```

Reply:
Not sure about the articles but I'd keep the "connection" singular since the exception is about a single connection.